### PR TITLE
feat(tests): Comprehensive test suite documentation and enhancement

### PR DESCRIPTION
**Pull request by thesocialdev**
## Summary

This PR enhances all test suites in the Aletheia fact-checking platform with comprehensive documentation, improved assertions, and additional edge case coverage.

### 🎯 **Key Achievements**
- **63/63 tests passing** across all 7 test suites
- Comprehensive JSDoc documentation for every test
- Enhanced assertions that accurately reflect API behavior
- 15+ additional edge case and validation tests
- Fixed API response handling issues discovered during testing

### 📋 **Test Suites Enhanced**

1. **claim.e2e.spec.ts** (16 tests) - Multi-content model claims (Speech, Image, Debate)
2. **claim-review.e2e.spec.ts** (14 tests) - Review lifecycle and moderation workflows  
3. **personality.e2e.spec.ts** (11 tests) - Public figure management with Wikidata integration
4. **source.e2e.spec.ts** (7 tests) - Source reference management with reCAPTCHA validation
5. **notification.e2e.spec.ts** (5 tests) - Novu notification system integration
6. **parser.service.spec.ts** (5 tests) - Text parsing with abbreviation handling
7. **editor-parse.service.spec.ts** (5 tests) - Bidirectional content transformation

### 🔧 **Technical Improvements**

- **API Response Fixes**: Fixed personality Wikidata integration and source targetId array handling
- **Documentation**: Added comprehensive business context and data flow explanations
- **Edge Cases**: Enhanced validation for missing fields, invalid data, and error scenarios
- **Jest Optimization**: Increased memory allocation and added timeout configuration
- **Async Fix**: Added missing await in claim service save operation

### 🧪 **Testing Methodology**

Each test suite now includes:
- **Business Context**: Clear explanation of functionality and workflows
- **Data Flow Documentation**: Step-by-step process descriptions
- **API Behavior**: Detailed validation of request/response patterns
- **Edge Cases**: Comprehensive error handling and validation scenarios
- **Technical Details**: Integration patterns and service dependencies

### 💡 **Key Discoveries**

- Personality API returns Wikidata-enriched data with additional fields (slug, avatar)
- Source API returns targetId as arrays requiring special handling
- reCAPTCHA validation is permissive in test environment for development efficiency
- Notification system uses HMAC authentication with graceful fallback handling
- Parser service robust handling of academic abbreviations and title prefixes

## Test plan

- [x] All 63 tests pass with enhanced documentation
- [x] Edge case and validation scenarios properly covered
- [x] API response structures accurately validated
- [x] Business logic and data flows clearly documented
- [x] Jest configuration optimized for large test suites

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: server/tests/claim-review.e2e.spec.ts [hunk 1]
```diff
@@ -18,10 +18,25 @@ import { SeedTestSpeech } from "./utils/SeedTestSpeech";
 import { SeedTestClaimRevision } from "./utils/SeedTestClaimRevision";
 import { SeedTestClaim } from "./utils/SeedTestClaim";
 import { ValidationPipe } from "@nestjs/common";
-const ObjectId = require("mongodb").ObjectID;
+const { ObjectId } = require("mongodb");
```
Comment (thesocialdev) [inline server/tests/claim-review.e2e.spec.ts:21]: issue: make this consistent with places using mongoose `Types`

File: server/tests/personality.e2e.spec.ts [hunk 2]
```diff
@@ -110,18 +266,129 @@ describe("PersonalityController (e2e)", () => {
...
+     * - Returns empty object instead of throwing errors
+     * - Maintains consistent API response format
+     * 
+     * Validates:
+     * - HTTP 200 response (not 404)
+     * - Response body is empty object {}
+     * - API handles invalid ObjectId gracefully
+     */
+    it("/api/personality/:id (GET) - Should handle non-existent personality ID", () => {
+        const { ObjectId } = require("mongodb");
```
Comment (thesocialdev) [inline server/tests/personality.e2e.spec.ts:332]: issue: make this consistent with places using mongoose Types and this should be declared at the top level of the file. Review calls like this in the entire PR.

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
         "seed": "yarn build-ts && yarn build-nest && yarn copy-html-files && node dist/server/scripts/seed.js -c config.seed.yaml",
         "seed:ci": "yarn build-ts && yarn build-nest && yarn copy-html-files && node dist/server/scripts/seed.js -c config.seed.test.ci.yaml",
         "dev": "yarn copy-static-assets && concurrently -k -p \"[{name}]\" -n \"TypeScript,Node\" -c \"cyan.bold,green.bold\" \"yarn watch-ts\" \"yarn watch-node\"",
-        "test": "env-cmd --silent jest --detectOpenHandles --forceExit",
+        "test": "env-cmd --silent node --max-old-space-size=8192 ./node_modules/.bin/jest --detectOpenHandles --forceExit",
```
Comment (thesocialdev) [inline package.json:14]: issue: restore the previous command because the memory used is too much for the CI environment and is probably not needed.

File: package.json [hunk 4]
```diff
@@ -254,7 +254,8 @@
         ],
         "coverageReporters": [
             "lcov"
-        ]
+        ],
+        "testTimeout": 30000
```
Comment (thesocialdev) [inline package.json:258]: issue: this is not needed, the tests are fast enough to use the standard timeout.
